nixos/modules/services/desktops/pipewire.nix

```diff
@@ -25,17 +35,76 @@ in {
           Automatically run pipewire when connections are made to the pipewire socket.
         '';
       };
+
+      alsa = {
+        enable = mkOption {
+          default = false;
+          type = types.bool;
+          description = ''
+            Route audio to/from generic ALSA-using applications via the ALSA PIPEWIRE PCM plugin.
+          '';
+        };
+
+        support32Bit = mkOption {
+          default = false;
+          type = types.bool;
+          description = ''
+            Whether to support sound for 32-bit ALSA applications on a 64-bit system.
+          '';
+        };
+      };
+
+      jack = {
+        enable = mkOption {
+          default = false;
+          type = types.bool;
+          description = ''
+            Enable transparent JACK audio emulation using LD_LIBRARY_PATH.
+          '';
+        };
+      };
+
+      pulse = {
+        enable = mkOption {
+          default = false;
+          type = types.bool;
+          description = ''
+            Enable transparent PulseAudio emulation using LD_LIBRARY_PATH.
+          '';
+        };
+      };
     };
   };
 
 
   ###### implementation
   config = mkIf cfg.enable {
-    environment.systemPackages = packages;
+    environment.systemPackages = [ pkgs.pipewire ]
+                                 ++ lib.optional cfg.jack.enable jack-libs
+                                 ++ lib.optional cfg.pulse.enable pulse-libs;
 
-    systemd.packages = packages;
+    systemd.packages = [ pkgs.pipewire ];
 
     systemd.user.sockets.pipewire.wantedBy = lib.mkIf cfg.socketActivation [ "sockets.target" ];
-  };
 
+    # If any paths are updated here they must also be updated in the package test.
+    sound.extraConfig = mkIf cfg.alsa.enable ''
+      pcm_type.pipewire {
+        libs.native = ${pkgs.pipewire.lib}/lib/alsa-lib/libasound_module_pcm_pipewire.so ;
+        ${optionalString enable32BitAlsaPlugins
+          "libs.32Bit = ${pkgs.pkgsi686Linux.pipewire.lib}/lib/alsa-lib/libasound_module_pcm_pipewire.so ;"}
+      }
+      pcm.!default {
+        @func getenv
+        vars [ PCM ]
+        default "plug:pipewire"
+        playback_mode "-1"
+        capture_mode "-1"
+      }
+    '';
+    environment.etc."alsa/conf.d/50-pipewire.conf" = mkIf cfg.alsa.enable {
+      source = "${pkgs.pipewire}/share/alsa/alsa.conf.d/50-pipewire.conf";
+    };
+    environment.sessionVariables.LD_LIBRARY_PATH = "/run/current-system/sw/lib/pipewire";
```

Comment:
Is there some dispatcher library, to which `runpath` we could add this, similar to `addOpenGLRunpath` works with libglvnd?

Reply:
Not currently. How exactly would you use such a package? With OpenGL the programs depend on it directly while the programs we're dealing with depend on PulseAudio or JACK instead. We could create some kind of audio `runpath` and use it for PulseAudio and JACK as well but I'm not sure whether such a large change would be worth it, at least for now.

Reply:
I will note however, that if we did do this it would allow for better support of both 32bit programs on 64bit machines as well as sandboxed applications.